From 5fccf13715879d8c6b91eaa77484b4a8de976f0a Mon Sep 17 00:00:00 2001
From: Igor Mammedov <imammedo@redhat.com>
Date: Tue, 13 Mar 2012 13:44:36 +0100
Subject: [PATCH 3/5] Make pause/resume_all_vcpus available to usage from
 hw/pc.c

RH-Author: Igor Mammedov <imammedo@redhat.com>
Message-id: <1331646277-28469-4-git-send-email-imammedo@redhat.com>
Patchwork-id: 38482
O-Subject: [RHEL6.3 qemu-kvm PATCH 3/4] Make pause/resume_all_vcpus available to usage from hw/pc.c
Bugzilla: 562886
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>

Bugzilla: 562886
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=4146795
Upstream made them public and moved them to cpus.[ch] in 296af7c9

Signed-off-by: Igor Mammedov <imammedo@redhat.com>
---
 cpu-all.h |    3 +++
 vl.c      |   11 ++++-------
 2 files changed, 7 insertions(+), 7 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 cpu-all.h |    3 +++
 vl.c      |   11 ++++-------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/cpu-all.h b/cpu-all.h
index 9c63a4a..65206e2 100644
--- a/cpu-all.h
+++ b/cpu-all.h
@@ -1101,4 +1101,7 @@ extern int64_t dev_time;
 void cpu_inject_x86_mce(CPUState *cenv, int bank, uint64_t status,
                         uint64_t mcg_status, uint64_t addr, uint64_t misc);
 
+void resume_all_vcpus(void);
+void pause_all_vcpus(void);
+
 #endif /* CPU_ALL_H */
diff --git a/vl.c b/vl.c
index 02fe0b2..5206005 100644
--- a/vl.c
+++ b/vl.c
@@ -3280,9 +3280,6 @@ static void vm_state_notify(int running, RunState state)
     }
 }
 
-static void resume_all_vcpus(void);
-static void pause_all_vcpus(void);
-
 void vm_start(void)
 {
     if (!runstate_is_running()) {
@@ -3646,14 +3643,14 @@ int qemu_cpu_self(void *env)
     return 1;
 }
 
-static void resume_all_vcpus(void)
+void resume_all_vcpus(void)
 {
     if (kvm_allowed) {
         kvm_resume_all_threads();
     }
 }
 
-static void pause_all_vcpus(void)
+void pause_all_vcpus(void)
 {
     if (kvm_allowed) {
         kvm_pause_all_threads();
@@ -3919,7 +3916,7 @@ static int all_vcpus_paused(void)
     return 1;
 }
 
-static void pause_all_vcpus(void)
+void pause_all_vcpus(void)
 {
     CPUState *penv = first_cpu;
 
@@ -3940,7 +3937,7 @@ static void pause_all_vcpus(void)
     }
 }
 
-static void resume_all_vcpus(void)
+void resume_all_vcpus(void)
 {
     CPUState *penv = first_cpu;
 
-- 
1.7.7.6

